controllers/release/adapter.go

```diff
@@ -1101,6 +1172,51 @@ func (a *adapter) createTenantPipelineRun(releasePlan *v1alpha1.ReleasePlan, sna
 	return pipelineRun, nil
 }
 
+// resolvePipelineReference resolves a git pipeline reference to a commit SHA and returns
+// both the resolved PipelineRef and the SHA. Returns the original ref if not a git resolver.
+// Any resolution failure will cause the release to be requeued.
+func (a *adapter) resolvePipelineReference(pipelineRef utils.PipelineRef, pipelineType string) (*tektonv1.PipelineRef, string, error) {
+	if pipelineRef.Resolver != "git" {
+		return pipelineRef.ToTektonPipelineRef(), "", nil
+	}
+
+	resolvedRef, err := pipelineRef.CreatePipelineRefWithResolvedSHA(a.ctx)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to resolve %s pipeline git reference to commit SHA: %w", pipelineType, err)
+	}
+
+	tektonRef := resolvedRef.ToTektonPipelineRef()
+	resolvedSHA, _ := resolvedRef.GetRevision()
+
+	a.logger.Info("Resolved pipeline reference to commit SHA",
+		"pipelineType", pipelineType,
+		"sha", resolvedSHA)
+
+	return tektonRef, resolvedSHA, nil
+}
+
+// resolveCollectorPipelineReference resolves a git pipeline reference to just the commit SHA.
+// This is used for collector pipelines that only need the SHA string.
+// Returns empty string for non-git resolvers.
+// Any resolution failure will cause the release to be requeued.
+func (a *adapter) resolveCollectorPipelineReference(pipelineRef utils.PipelineRef, collectorType string) (string, error) {
```

Comment:
Same here for the `collectorType` arg.

Reply:
the same idea: it's better for debuging. we always know related to the failure collector type